src/main/java/dev/jbang/cli/Alias.java

```diff
@@ -79,13 +74,15 @@ public Integer doCall() {
 					"Invalid alias name, it should start with a letter followed by 0 or more letters, digits, underscores or hyphens");
 		}
 		try {
-			Script script = BaseScriptCommand.prepareScript(scriptOrFile);
+			ExtendedRunUnit script = RunUnit.forResource(scriptOrFile);
 			if (name == null) {
 				name = AppInstall.chooseCommandName(script);
 			}
 
-			if (description == null && script.getDescription() != null && script.getDescription().length() > 0) {
-				description = script.getDescription();
+			if (description == null && script.runUnit instanceof Script) {
+				if (script.script().getDescription() != null && script.script().getDescription().length() > 0) {
```

Comment:
maybe rename `script` to `unit` ?

Reply:
so its unit.script()

Reply:
Yeah, but didn't feel like renaming all those yet :-)
Most of that is manual work, no good way to do a search & replace.

Reply:
I renamed all of them to `xrunit`. Not a beautiful name but it's one that's very easy to search & replace for a future rename :-)